File: data/cpp/tests/test_Polars_mean.cpp

#include <gtest/gtest.h>
#include "../src/function_Polars_mean.cpp"

class TestPolarsExample : public ::testing::Test {
protected:
    PolarsExample calc;
};

TEST_F(TestPolarsExample, TestComputeMeanMultiplePositiveNumbers) {
    auto result = calc.compute_mean(1, 2, 3);
    ASSERT_TRUE(result);
    EXPECT_FLOAT_EQ(*result, 2.0f);
}

TEST_F(TestPolarsExample, TestComputeMeanMultipleNegativeNumbers) {
    auto result = calc.compute_mean(-1, -2, -3);
    ASSERT_TRUE(result);
    EXPECT_FLOAT_EQ(*result, -2.0f);
}

TEST_F(TestPolarsExample, TestComputeMeanMixedNumbers) {
    auto result = calc.compute_mean(1, -1, 2);
    ASSERT_TRUE(result);
    EXPECT_FLOAT_EQ(*result, 0.6666666666666666f);
}

TEST_F(TestPolarsExample, TestComputeMeanSingleNumber) {
    auto result = calc.compute_mean(10);
    ASSERT_TRUE(result);
    EXPECT_FLOAT_EQ(*result, 10.0f);
}

TEST_F(TestPolarsExample, TestComputeMeanNoInput) {
    auto result = calc.compute_mean(10, boost::none, boost::none);
    ASSERT_TRUE(result);
    EXPECT_FLOAT_EQ(*result, 10.0f);
}

int main(int argc, char **argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/java/tests/test_Polars_mean.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class test_Polars_mean {

    @Test
    public void test_compute_mean_multiple_positive_numbers() {
        function_Polars_mean calc = new function_Polars_mean();
        Double result = calc.compute_mean(1f, 2f, 3f);
        assertEquals(2.0, result, 0.0001);
    }

    @Test
    public void test_compute_mean_multiple_negative_numbers() {
        function_Polars_mean calc = new function_Polars_mean();
        Double result = calc.compute_mean(-1f, -2f, -3f);
        assertEquals(-2.0, result, 0.0001);
    }

    @Test
    public void test_compute_mean_mixed_numbers() {
        function_Polars_mean calc = new function_Polars_mean();
        Double result = calc.compute_mean(1f, -1f, 2f);
        assertEquals(0.6666666666666666, result, 0.0001);
    }

    @Test
    public void test_compute_mean_single_number() {
        function_Polars_mean calc = new function_Polars_mean();
        Double result = calc.compute_mean(10f, null, null);
        assertEquals(10.0, result, 0.0001);
    }

    @Test
    public void test_compute_mean_no_input() {
        function_Polars_mean calc = new function_Polars_mean();
        Double result = calc.compute_mean(10f, null, null);
        assertEquals(10.0, result, 0.0001);
    }
}